### PR TITLE
[receiver/riak] - Fix issue where user configured metric settings were ignored

### PR DESCRIPTION
**Pull request by armstrmi**
**Description:** Fixing minor bug, issue with Riak receiver newScraper method, would only implement `DefaultMetricSettings`

**Link to tracking Issue:** 
(https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/9560)

**Testing:** 

**Documentation:** <Describe the documentation added.>

### CODE REVIEW

**Review by dehaansa — COMMENTED** [review]
Looks like you should be able to add a relatively straightforward test in your scraper tests with some metrics disabled to validate the fix. 👍

**Review by dehaansa — APPROVED** [review]

File: receiver/riakreceiver/scraper_test.go [hunk 1]
```diff
@@ -90,6 +92,7 @@ func TestScaperScrape(t *testing.T) {
 		desc              string
 		setupMockClient   func(t *testing.T) client
 		expectedMetricGen func(t *testing.T) pmetric.Metrics
+		metricsEnabled    bool
```
Comment (djaglowski) [inline receiver/riakreceiver/scraper_test.go:95]: This is a very specific test case, so we shouldn't enhance the framework to consider it specifically. A better way is to allow each test case to provide the config it needs with a similar mechanism to `setupMockClient` or `expectedMetricsGen`.

File: CHANGELOG.md [hunk 2]
```diff
@@ -40,6 +40,7 @@
 - `resourcedetectionprocessor`: Wire docker detector (#9372)
 - `kafkametricsreceiver`: The kafkametricsreceiver was changed to connect to kafka during scrape, rather than startup. If kafka is unavailable the receiver will attempt to connect during subsequent scrapes until succcessful (#8817).
 - `datadogexporter`: Update Kubernetes example manifest to new executable name. (#9425).
+- `riakreceiver`: Updated newScraper method to implement given configs Metric Settings. (#9561)
```
Comment (djaglowski) [inline CHANGELOG.md:43]: This isn't very clear to a user.
```suggestion
- `riakreceiver`: Fix issue where user configured metric settings were ignored. (#9561)
```

**Review by djaglowski — COMMENTED** [review]
@armstrmi, a couple minor points of feedback. Also, please note that issues and pull requests in this repository [should be titled](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/CONTRIBUTING.md#pull-request-title) as `[receiver/riak] - Fix issue where user configured metric settings were ignored`

File: receiver/riakreceiver/testdata/scraper/expected_disabled.json [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+                   "key": "riak.node.name",
+                   "value": {
+                      "stringValue": "riak@127.0.0.1"
+                   }
+                }
+             ]
+          }
+       }
+    ]
+ }
```
Comment (djaglowski) [inline receiver/riakreceiver/testdata/scraper/expected_disabled.json:162]:
```suggestion
 }
 
```

File: receiver/riakreceiver/scraper_test.go [hunk 4]
```diff
@@ -156,3 +193,26 @@ func TestScaperScrape(t *testing.T) {
...
+			Enabled: true,
+		},
+		RiakVnodeIndexOperationCount: metadata.MetricSettings{
+			Enabled: true,
+		},
+		RiakVnodeOperationCount: metadata.MetricSettings{
+			Enabled: true,
+		},
+	}
+}
```
Comment (djaglowski) [inline receiver/riakreceiver/scraper_test.go:218]: This is creating a specific scenario. Now that we have a `setupCfg` setting for each test, just do this in the specific test that requires this.

File: receiver/riakreceiver/testdata/scraper/expected_disabled.json [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+
```
Comment (djaglowski) [inline receiver/riakreceiver/testdata/scraper/expected_disabled.json:163]: The file needs to end with a newline, not a newline with a space.
```suggestion

```

**Review by djaglowski — APPROVED** [review]

**Pull request merged**